### PR TITLE
Pete / Burr / Sande - here are my two quickstarts (mail, logging) - check them out and let me know what you would like modified, etc

### PR DESCRIPTION
**Pull request by jtosi**
Modifications done per Pete's request.  Let me know what else you would like

ADDED LOGGING.PROPERTIES FILE FOR EXAMPLE
ADDED JAVA CLASS FILE FOR LOGGING
ADDED JSP PAGE TO FORCE LOGGING
ADDED SAMPLE STANDALONE.XML FOR LOGGING CONFIGURATION
ADDED README FILES FOR MAIL QUICKSTART
ADDED EMAIL OBJECT FOR MAIL QUICKSTART
ADDED JSF FRONTED FOR SENDING EMAIL

### CODE REVIEW

File: logging/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Logging Example
```
Comment (pmuir) [inline logging/README.md]: Joel, we're trying to stick to a standard layout for the READMEs - if you look at e.g. numberguess or kitchensink you can see the headings we normally use. Would you mind updating?

Excellent description :-)

File: logging/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+===================
+
+This example demonstrates how to setup and log different levels of information in *JBoss AS 7*.
+
+The example contains just one class file and one jsp file that, once accessed, fires off the logging information.
+Also included in the example are an updated standalone.xml and logging.properties files.  Both of these would go in your
+*JBOSS_HOME*/standalone/configuration directory.  An example of asynchronous logging is included in the configuration.
+Be sure to make a backup of your original files beforehand.
```
Comment (pmuir) [inline logging/README.md]: I would suggest we make this a step-by-step process

e.g.

Step 1, fire up the app server and deploy the example. See what log messages you get when running it.
Step 2, copy XXX to YYY
Step 3, restart the app server and observe how the log messages have changed.

File: logging/logging.properties [hunk 3]
```diff
@@ -0,0 +1,101 @@
```
Comment (pmuir) [inline logging/logging.properties:0]: Could you add some comments to indicate how this differs from the defaults?

File: logging/pom.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+   </dependencyManagement>
+
+   <dependencies>
+
+      <!-- Import the CDI API, we use provided scope as the API is included 
+         in JBoss AS 7 -->
+
+      <!-- Import the Common Annotations API (JSR-250), we use provided scope 
+         as the API is included in JBoss AS 7 -->
+
```
Comment (pmuir) [inline logging/pom.xml]: Could you remove these two comments?

File: logging/src/main/webapp/WEB-INF/beans.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<!-- Marker file indicating CDI 1.0 should be enabled -->
+
+<beans xmlns="http://java.sun.com/xml/ns/javaee"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xsi:schemaLocation="
+      http://java.sun.com/xml/ns/javaee 
+      http://java.sun.com/xml/ns/javaee/beans_1_0.xsd">
+</beans>
```
Comment (pmuir) [inline logging/src/main/webapp/WEB-INF/beans.xml]: You could remove this file as you aren't using CDI.

File: logging/src/main/webapp/WEB-INF/faces-config.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<!-- Marker file indicating JSF 2.0 should be enabled in the application -->
+
+<faces-config version="2.0"
+   xmlns="http://java.sun.com/xml/ns/javaee"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xsi:schemaLocation="
+      http://java.sun.com/xml/ns/javaee
+      http://java.sun.com/xml/ns/javaee/web-facesconfig_2_0.xsd">
+      
+</faces-config>
```
Comment (pmuir) [inline logging/src/main/webapp/WEB-INF/faces-config.xml]: You could remove this file as you aren't using JSF.

File: logging/src/main/webapp/home.xhtml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+         </div>
+         <div>
+            <h:commandButton id="restartButton" value="Reset" type="reset"/>
+         </div>
+      </h:form>
+   </div>
+   <br style="clear: both" />
+</body>
+</html>
\ No newline at end of file
```
Comment (pmuir) [inline logging/src/main/webapp/home.xhtml]: You should remvoe this file as it seems unrelated to the logging quickstart.

File: mail/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+E-Mail Example
```
Comment (pmuir) [inline mail/README.md]: As above, should follow the standard layout for readmes.

File: mail/src/main/java/org/jboss/as/quickstarts/mail/Email.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+public class Email implements Serializable {
+	
+	private static final long serialVersionUID = 1544680932114626710L;
+	
+	/**
+	 * Resource for sending the email.  The mail subsystem is defined in either standalone.xml or domain.xml in your 
+	 * respective configuration directory. 
+	 */
+	@Resource (mappedName="java:jboss/mail/Default")
+	private Session mySession;
```
Comment (pmuir) [inline mail/src/main/java/org/jboss/as/quickstarts/mail/Email.java:42]: I know it seems silly for such a trivial example, but you might want to use a Resources class like:

```
public class Resources {

    @Produces
    @Resource (mappedName="java:jboss/mail/Default")
    private Session mySession;

} 
```

As then you can just @Inject the Session wherever you want.

File: mail/src/main/webapp/WEB-INF/web.xml [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+ </servlet-mapping>
+ <welcome-file-list>
+  <welcome-file>index.html</welcome-file>
+  <welcome-file>index.htm</welcome-file>
+  <welcome-file>index.jsp</welcome-file>
+  <welcome-file>default.html</welcome-file>
+  <welcome-file>default.htm</welcome-file>
+  <welcome-file>default.jsp</welcome-file>
+ </welcome-file-list>
+</web-app>
```
Comment (pmuir) [inline mail/src/main/webapp/WEB-INF/web.xml]: We shouldn't need this file. Can you try without it?

File: logging/README.md (hunk 1 above)
Comment (jtosi) [inline logging/README.md]: Hey Pete - are we no longer doing the HTML versions?  I looked at a few others and that appears to be the case, just wanted to make sure.

Comment (pmuir) [inline logging/README.md]: HTML versions of the READMEs?

They are now generated during a release.

Comment (jtosi) [inline logging/README.md]: Ahh, cool. I didn't know that.  Guess I should read email more ;)

Thanks dude

On Tue, Feb 28, 2012 at 7:57 AM, Pete Muir
reply@reply.github.com
wrote:

File: mail/src/main/java/org/jboss/as/quickstarts/mail/Email.java (hunk 9 above)
Comment (jtosi) [inline mail/src/main/java/org/jboss/as/quickstarts/mail/Email.java:42]: Hey dude, I go back and forth on this one.  To me, the quickstarts are about 'the simplest thing that works'.  To me, adding an extra class kind of detracts from that. Are you ok with not adding this?  Your call boss

Comment (pmuir) [inline mail/src/main/java/org/jboss/as/quickstarts/mail/Email.java:42]: On 28 Feb 2012, at 15:11, Joel Tosi wrote:

> > - \* @author Joel Tosi
> > - \* 
> > - */
> >   +
> >   +@Named
> >   +@SessionScoped
> >   +public class Email implements Serializable {
> > -   
> > -    private static final long serialVersionUID = 1544680932114626710L;
> > -   
> > -    /**
> > -     \* Resource for sending the email.  The mail subsystem is defined in either standalone.xml or domain.xml in your 
> > -     \* respective configuration directory. 
> > -     */
> > -    @Resource (mappedName="java:jboss/mail/Default")
> > -    private Session mySession;
> 
> Hey dude, I go back and forth on this one.  To me, the quickstarts are about 'the simplest thing that works'.  To me, adding an extra class kind of detracts from that. Are you ok with not adding this?  Your call boss

I'm happy either way.

File: logging/README.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+The example contains just one class file and one jsp file that, once accessed, fires off the logging information.
+Also included in the example are an updated standalone.xml and logging.properties files.  
+
+Follow these steps to better visualize the logging working:
+
+1) First deploy the jsp file and access it at <http://localhost:8080/jboss-as-logging>
+
+2) Look at your log files, located at *JBOSS_HOME*/standalone/log.  What information do you see there?  What came from the application?
+
+3) Copy the provided standalone.xml and logging.properties files.  Both of these files will go in your *JBOSS_HOME*/standalone/configuration directory.  Be sure to make a backup of your original files beforehand.
```
Comment (pmuir) [inline logging/README.md]: I think we have a problem with this step. `standalone.xml` changes quite a lot, so we're going to have a hell of a time keeping it in sync. Is there a way to do it without having the user copy these files in?

File: logging/README.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+  
+4) Restart your server
+
+5) Access the jsp at <http://localhost:8080/jboss-as-logging>
```
Comment (pmuir) [inline logging/README.md]: I think you are missing the index.html to bounce you to the home.jsp?

**Comment by pmuir** [comment]
Joel - a few fixes for you to merge in - https://github.com/pmuir/quickstart/commit/696317476d94de42d484d379c0ab0051a109fa2b

**Comment by sgilda** [comment]
Hi Joel,
I am testing your logging quickstart against JBoss 7.1.1 and have run into a problem. I'm sure it's something stupid that I'm doing.

First, in the README file, step 1 says to deploy the quickstart. Although it seems intuitive, I think you need to tell them to start the server first. I wasn't clear whether I needed to start the server with the full profile or not. So I started the server without it: JBOSS_HOME/bin/standalone.sh.

I built and deployed the logging quickstart. I see this in the server console, so it appears to deploy OK: 
10:43:59,577 INFO  [org.jboss.as.server.deployment](MSC service thread 1-3) JBAS015876: Starting deployment of "jboss-as-logging.war"
10:44:00,055 INFO  [org.jboss.web](MSC service thread 1-3) JBAS018210: Registering web context: /jboss-as-logging
10:44:00,212 INFO  [org.jboss.as.server](management-handler-thread - 2) JBAS018559: Deployed "jboss-as-logging.war"

Yet, when I hit the URL: http://localhost:8080/jboss-as-logging/, I get an HTTP Status 404:
HTTP Status 404 - /jboss-as-logging/
type Status report
message /jboss-as-logging/
description The requested resource (/jboss-as-logging/) is not available.

When I go to the console, http://localhost:9990/console/App.html#deployments, I see jboss-as-logging.war and it's enabled.

Do you have any idea what I'm doing wrong?

Thanks, Sande

**Comment by sgilda** [comment]
Looking ahead in the README, I have a few comments

2) Look at your log files, located at _JBOSS_HOME_/standalone/log.  What information do you see there?  What came from the application?
It might be nice to answer the question. How will they know if they are getting the right result? ;-)

3) Copy the provided standalone.xml and logging.properties files.  Both of these files will go in your _JBOSS_HOME_/standalone/configuration directory.  Be sure to make a backup of your original files beforehand.

You should lead with "Make a backup  copy of the standalone.xml and logging.properties files located in your JBOSS_HOME/standalone/configuration/ directory." Some of us don't read ahead. ;-)

Also, instead of replacing the standalone.xml file, would it be possible to provide XML snippets to copy in along with instructions as to where they go? When I do a diff on your standalone.xml file and the one I have, there are a ton of differences and I'm pretty sure not all are related to changes needed for this quickstart. If there are too many changes, it would be nice to have some sort of summary of what needs to be added to these files to run the example.

The logging properties are clearer. I like the markers for the START and END, but it would still be nice to provide the changes and have them copy them into the existing properties file. It forces them to look at them.

**Comment by sgilda** [comment]
The mail quickstart works great! 
The only comments I have are that you might want to give instructions to start the server before deployment. :-)

**Comment by pmuir** [comment]
I've merged in the mail quickstart only. Logging needs though as discussed.

**Comment by sgilda** [comment]
Are we still waiting for fixes to the logging quickstart for thiis?

**Comment by jtosi** [comment]
Yeah. Sorry about that.  They have been sitting in the middle of my queue,
other items kept popping up.  I will get to them by Saturday at the latest.

Thanks for the reminder
On Apr 4, 2012 7:18 AM, "sgilda" <
reply@reply.github.com>
wrote:

**Comment by jtosi** [comment]
I didn't make it to this this weekend - foot injury on Friday.  Just
wanted to let you know I hadn't forgotten

**Comment by sgilda** [comment]
Hi Joel,
Just checking in to see how things are going with the logging quickstart changes.
Thanks,
Sande

**Comment by jtosi** [comment]
Morning Sande,
   I am actually working on this now. I am hoping to wrap it up this
morning.  Please correct me if I am wrong, but it is just the logging one
that needs changes, correct?

best,
Joel

On Fri, May 4, 2012 at 6:45 AM, sgilda <
reply@reply.github.com

File: logging/README.md (hunk 11 above)
Comment (jtosi) [inline logging/README.md]: Yeah, modifying standalone.xml is going to be a bad route, and not even the recommended approach.  Do you know if the CLI would allow me to make those changes - modifying log settings?  If so, I could write a script to do it, but then we are expecting the user to be able to execute the CLI script.  Is that any better?

File: logging/README.md (hunk 12 above)
Comment (jtosi) [inline logging/README.md]: Yep, that was it.  I put it back in, added a message to the body of the HTML so the user would know the log messages were written.

**Comment by jtosi** [comment]
Added the index.html for the redirect and addressed the readme's.  Hopefully this is good.  Committing back now

**Comment by jtosi** [comment]
Should be set now (I hope)

Thanks for the patience

File: logging/README.md (hunk 11 above)
Comment (pmuir) [inline logging/README.md]: Yes, I think you could do this via the CLI fine.

**Comment by sgilda** [comment]
Hi Joel,
Thanks for doing the updates. I'm in the middle of a merge at the 
moment, but will try to get to this soon.
Thanks again,
Sande

File: logging/README.md (hunk 11 above)
Comment (jtosi) [inline logging/README.md]: Pete - I'm honestly not going to have time to write a CLI script for this,
as small as it is, for a few weeks.  I don't want to be any more of a
bottleneck than I already have been.

Should we go with the documentation as is (significant updates to it)?

Tosi

On Mon, May 7, 2012 at 12:15 PM, Pete Muir <
reply@reply.github.com

**Comment by sgilda** [comment]
I'm not sure how extensive the changes are to your standalone.xml file, but some of the other quickstarts provide XML snippets with instructions where to copy it into the standalone.xml file. 

For example:

Find the logging subsystem <subsystem xmlns="urn:jboss:domain:logging:1.1"> and copy the following XML after the <console-handler> element:

```
        <!-- EXAMPLE FILE LOGGER CONFIGURATION FOR QUICKSTART WARNING-->
        <periodic-rotating-file-handler name="FILE_QS_WARN">
            <!-- THE LEVEL WE WANT TO LOG-->
            <level name="WARN"/>
            <!-- FORMATTING OF THE LOG MESSAGES (date, message) -->
            <formatter>
                <pattern-formatter pattern="%d{HH:mm:ss,SSS} %-5p [%c] (%t) %s%E%n"/>
            </formatter>
            <!-- WHERE THIS LOG SHOULD BE WRITTEN TO-->
            <file relative-to="jboss.server.log.dir" path="quickstart.warn.log"/>
            <!-- WHEN THE LOGS ROTATE, THE SUFFIX TO ADD TO THE FILENAME-->
            <suffix value=".yyyy-MM-dd"/>
            <!-- APPEND TO THE END OF FILES-->
            <append value="true"/>
        </periodic-rotating-file-handler>
        <!-- EXAMPLE FILE LOGGER CONFIGURATION FOR QUICKSTART ERRORS-->
        <periodic-rotating-file-handler name="FILE_QS_ERROR">
            <level name="ERROR"/>
            <formatter>
                <pattern-formatter pattern="%d{HH:mm:ss,SSS} %-5p [%c] (%t) %s%E%n"/>
            </formatter>
            <file relative-to="jboss.server.log.dir" path="quickstart.error.log"/>
            <suffix value=".yyyy-MM-dd"/>
            <append value="true"/>
        </periodic-rotating-file-handler>
```

Is something like this feasible?

**Comment by sgilda** [comment]
Joel, there have been a number of changes to the format of the README files since you did this. I will modify your  file and email it to you to save you some time. 
Thanks,
Sande

**Comment by sgilda** [comment]
Hi Joel,
I emailed you an updated README.md file. I wasn't able to verify the accuracy of my instructions because even though it deployed successfully, I wasn't able to access the application.

I can see this in the server log:
16:34:11,948 INFO  [org.jboss.as.server.deployment](MSC service thread 1-2) JBAS015876: Starting deployment of "jboss-as-logging.war"
16:34:12,721 INFO  [org.jboss.web](MSC service thread 1-1) JBAS018210: Registering web context: /jboss-as-logging
16:34:12,881 INFO  [org.jboss.as.server](management-handler-thread - 2) JBAS018559: Deployed "jboss-as-logging.war"

Yet, when I access http://localhost:8080/jboss-as-logging-tools/ (or http://localhost:8080/jboss-as-logging-tools/home.jsf), I get an HTTP Status 404. I'm not sure what what I'm doing wrong. Do you have any ideas?

You will also need to update your pom.xml file to use the new BOMs. Take a look at the other quickstarts and you'll see that we now define properties for the BOM version and use them in the dependencies section.

You'll also need to copy in the copyright comments at the beginning of the files. You can find examples in the other quickstarts.

Thanks,
Sande

**Comment by sgilda** [comment]
Hi Joel,

Looks great! I hope you made sure I didn't change the true meaning of anything. :-)

I discovered why I'm getting the "HTTP status 404: /jboss-as-logging/ error." There is no redirect to home.jsp. I can run the quickstart if I type in the full URL: http://localhost:8080/jboss-as-logging/home.jsp

We would like all the quickstarts to be accessible using the root URL  I believe you can easily fix this problem by renaming home.jsp to index.jsp. Or, you could add an index.html file that does a redirect to that page. (see https://github.com/jboss-jdf/jboss-as-quickstart/blob/master/numberguess/src/main/webapp/index.html

Also, for all the quickstarts where the user needs to modify the standalone.xml configuration file, I'm trying to add the alternative jboss-cli commands in addition to the copy/paste XML snippents. Would you happen to know the jboss-cli commands to modify the <subsystem xmlns="urn:jboss:domain:logging:1.1"> for this quickstart?

Thanks,
Sande

**Comment by sgilda** [comment]
Sent an email reminder to Joel.

**Comment by sgilda** [comment]
Sent another email reminder 2012/11/02.

**Comment by sgilda** [comment]
20121217: Sent email reminder to Joel.

**Comment by sgilda** [comment]
Received an email from @jtosi on April 5, 2013 asking if his pull is still active. I assured him it still lis. :-)

My response:
The redirect is minor and there are plenty of examples of jboss-cli commands in the quickstarts. You can follow the template in servlet-security, helloworld-jms, or jts. We give instructions to configure the server using the JBoss CLI first, then manually.

**Comment by sgilda** [comment]
@jtosi : Just checking to see if you've had time to look at these quickstarts. :-)

**Comment by sgilda** [comment]
I have issued a new pull for this one here: https://github.com/jboss-jdf/jboss-as-quickstart/pull/521

Closing this one as it is outdated.
